core/startos/src/service/service_effect_handler.rs

```diff
@@ -310,11 +321,44 @@ struct MountParams {
     location: String,
     target: MountTarget,
 }
+async fn set_system_smtp(
+    context: EffectContext,
+    data: SetSystemSmtpParams,
+) -> Result<(), Error> {
+    let context = context.deref()?;
+    context
+        .ctx
+        .db
+        .mutate(|db| {
+            let model = db.as_public_mut()
+            .as_server_info_mut()
+            .as_smtp_mut();
+            model.ser(&mut Some(data.smtp))
+        })
+        .await
+}
 async fn get_system_smtp(
     context: EffectContext,
     data: GetSystemSmtpParams,
-) -> Result<Value, Error> {
-    todo!()
+) -> Result<String, Error> {
```

Comment:
is this really just a string?

Reply:
Good catch. This should follow the TS type defined [here](https://github.com/Start9Labs/start-os/blob/next/minor/sdk/lib/types.ts#L153-L159).

I will update pending clarification whether this should be an os rpc call similar to `setSystemRpc`